server/s3_put_object_action_base.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <string>
+
+#include <gtest/gtest_prod.h>
+
+#include "lib/types.h"  // struct m0_uint128
+#include "s3_object_action_base.h"
+
+enum class S3PutObjectActionState {

```

Comment:
should we rename this enum class as S3ObjectActionState, so that it would be generic and can be sued in any Action class ?

Reply:
Not sure. This base class can be applied to 4 actions: put object, put multipart, put chuck and copy objects.
All are PUT. So i decided to name the enum as is.